hydra-cluster/test/Test/CardanoNodeSpec.hs

```diff
@@ -39,18 +66,18 @@ spec = do
         slot2 <- queryTipSlotNo networkId nodeSocket
         slot2 `shouldSatisfy` (> slot1)
 
-    describe "findRunningCardanoNode" $ do
-      it "returns Nothing on non-matching network" $ \(tr, tmp) -> do
-        withCardanoNodeOnKnownNetwork tr tmp Preview $ \_ -> do
+  describe "findRunningCardanoNode" $ do
+    forASupportedQuickNetwork "returns Nothing on non-matching network" $ \network -> do
+      setupTracerAndTempDir $ \(tr, tmp) ->
+        withCardanoNodeOnKnownNetwork tr tmp network $ \_ -> do
           findRunningCardanoNode tr tmp Preproduction `shouldReturn` Nothing
 
-      it "returns Just running node on matching network" $ \(tr, tmp) -> do
-        withCardanoNodeOnKnownNetwork tr tmp Preview $ \runningNode -> do
-          findRunningCardanoNode tr tmp Preview `shouldReturn` Just runningNode
+    forASupportedQuickNetwork "returns Just running node on matching network" $ \network -> do
+      setupTracerAndTempDir $ \(tr, tmp) ->
+        withCardanoNodeOnKnownNetwork tr tmp network $ \runningNode -> do
+          findRunningCardanoNode tr tmp network `shouldReturn` Just runningNode
```

Comment:
Should: not need to distinguish quick and non-quick networks here.

`findRunningCardanoNode` is expected to work on any of the supported networks. `Preview` and `Production` were just arbitrarily used before.

Reply:
It actually doesn't. It works if you use a threadDelay on Preproduction, but it still failed for Mainnet after ten seconds.

Reply:
Why not?

(I have a suspicion it being `Byron` era incompatibility, but want to have evidence)

It can't be because of "quickness" because Preprod and preview have the same block production parameters and nothing is quicker on them than on mainnet really.

Reply:
Here's a run where it fails for Preprod and Mainnet. https://github.com/input-output-hk/hydra/actions/runs/9206469624/job/25324644966?pr=1448

Reply:
```
{"err":"QueryNotShelleyBasedEraException (AnyCardanoEra ByronEra)","tag":"MsgQueryGenesisParametersFailed"}}
```

So this is the root issue.

Reply:
What do we do about this?